test/fixtures/ast/custom-template-tokenizer/token-ranges.json

```diff
@@ -0,0 +1,26 @@
+[
+    "<template",
+    "lang",
+    "=",
+    "\"customLang\"",
+    ">",
+    "\n    ",
+    "A-totally",
+    ":",
+    "made",
+    "=",
+    "\"",
+    "up",
+    "\"",
+    "]",
+    "{{",
+    "templating",
+    "+",
+    "language",
+    "}}",
+    "",
```

Comment:
I think a token with no width can cause an infinite loop.

For example, if you do the following.

```js
   let t = token
    while ( (t = tokenStore.getTokenBefore(t)) != null ) {
      // process for t
    }
```

Reply:
I haven't encountered an infinite loop with my actual pug tokenizer, but I can remove empty tokens from the mock tokenizer in the test.